fdbmonitor/fdbmonitor.cpp

```diff
@@ -751,6 +753,10 @@ void start_process(Command* cmd, ProcessID id, uid_t uid, gid_t gid, int delay,
 		signal(SIGHUP, SIG_DFL);
 		signal(SIGINT, SIG_DFL);
 		signal(SIGTERM, SIG_DFL);
+#ifdef __linux__
+		signal(SIGCHLD, SIG_DFL);
+#endif
+		sigprocmask(SIG_SETMASK, mask, nullptr);
```

Comment:
To avoid changing the behavior for MacOS, should this be included in __freebsd__?

Reply:
Imho, executing sigprocmask() only under linux is a typo.

Reply:
It looks like the goal of this is to restore the original signal mask to the child processes, which in theory could run on all platforms. It does also seem that we don't actually change the signal mask except on Linux, so maybe this doesn't actually need to be run on anything but Linux. If we wanted to do the latter, though, it may be clearer to have this function call `sigprocmask` unconditionally and then have the other platforms pass in a `null` mask.

Reply:
@sfc-gh-abeamon, I believe, there's a moment when the signal mask can be changed on non-Linux system. `load_conf()` via `start_process()` is executed with `&normal_mask` on all platforms at https://github.com/apple/foundationdb/blob/713fce36572ffc3ac24c1275bfc4f36240709886/fdbmonitor/fdbmonitor.cpp#L1628